### DISCUSSION TITLE
Trouble building from source

### DISCUSSION THREAD

**Discussion by loserprance** [nirgoren/Eddienput#30] [Technical support]
I'm trying to run Eddienput built from the source code and have been unsuccessful so far. I wasn't sure if I should make this an issue, but I don't interpret it as a problem with the code itself more than it is my limited understanding of how to build from source.

The build.bat process completes smoothly but running the .exe gives me a fatal error about failing to execute the script GUI; I've tried copying all the Qt related files from the latest release folder into the dist directory but still get the same result; copying _all_ files from release that don't exist in dist fails to launch with no error.

I feel like this isn't the right way to go about it? Any suggestions?

**Comment by nirgoren** [comment]
The expected Python version for this project is currently 3.7. Version 3.8 introduced some [changes](https://docs.python.org/3/library/os.html#os.add_dll_directory) to how dlls are loaded. The bat script uses pyinstaller, which doesn't seem to work well together with some third party libraries that are used in the project due to this change if run with Python 3.8 and higher.

The main branch bat file now explicitly states running pyinstaller with Python 3.7.

Note that building is not neccesary for running the program. With a 3.7 python environment set up with all the required dependencies (PyQt5, pynput, XInput-Python) installed, running the `gui.py` script from the `src` folder should start up the program.

**Reply by murkilator** [reply]
Running gui.py directly fails because when it loads the default recording it needs game configs to be in Eddienput/src/configs, but that doesn't exist. When I copied Eddienput/configs into to Eddienput/src it worked
